Engine/Server/Src/GameEngine/Application/Application.cpp

```diff
@@ -139,8 +142,30 @@ GameEngine::Application::Application()
     }, *_registries);
 }
 
+bool GameEngine::Application::noPlayerConnected()
+{
+    ECS::SparseArray<IComponent> PlayerComponentArray = _registries->get_components<IComponent>("PlayerComponent");
+    for (std::size_t index = 0; index < PlayerComponentArray.size(); index++) {
+        PlayerComponent* player = dynamic_cast<PlayerComponent*>(PlayerComponentArray[index].get());
+        if (player && player->token != 0) {
+            return false;
+        }
+    }
+    return true;
+}
+
 void GameEngine::Application::run()
 {
+
+    while (noPlayerConnected()) {
+#ifdef _WIN32
+        Sleep(0.5);
+#else
+        sleep(0.5);
+#endif
```

Comment:
I think you should define a macro in a .hpp file instead of using #ifdef and #endif in a cpp file

Reply:
I can't, it's for a function.
If I create a Macro, both functions will be compiled.

Reply:
You can't do this in your .hpp ?
```suggestion
#ifdef _WIN32
        #define SLEEP(x) Sleep(x)
#else
        #define SLEEP(x) sleep(x)
#endif
```

Reply:
Strange but fine

Reply:
Less strange than using ifdef and endif in hpp for me x)